### ISSUE TITLE
Getting the deps for a single commit is overly complicated

### ISSUE THREAD

**Issue by mstefani** [aspiers/git-deps#82]
I have been using until now git-deps 0.1.0-15-gbd9ed69.
With that `git deps commit` would get the dependencies just for `commit`.

That changed with

```
commit d601e35f6ea42eb6110e3d1449d03f16a1da3f10
Author: Adam Spiers <git@adamspiers.org>
Date:   Thu May 26 00:28:25 2016 +0100

    allow passing a revision range to CLI or web UI
```

With that `git deps commit` gets now the dependencies for `commit` and *all* older commits.
To get now the deps just for `commit` one has to use the unwieldy:
`git deps commit^..commit`

**Comment by aspiers** [comment]
Yes, it does expect a range now.  It's worth noting that you can type

    git deps commit^!

which is shorter than

    git deps commit^..commit

Having said that, there is a proposal in #67 for making this more user-friendly, but I haven't got around to figuring out how to finish it yet.

**Comment by midenok** [comment]
What --recurse option means then? I guess if single commit specified it should work like for single commit, not for infinity..commit. Duplicates #90.

**Comment by aspiers** [comment]
[**@midenok**](https://github.com/midenok) commented on [April 18, 2019 12:30 PM](https://github.com/aspiers/git-deps/issues/82#issuecomment-484464814 "2019-04-18T11:30:02Z - Replied by Github Reply Comments"):
> What --recurse option means then? I guess if single commit specified it should work like for single commit, not for -Infinity..commit. Duplicates #90.

`--recurse` is different, because then it recurses through discovered dependencies rather than only analysing the commits mentioned on the CLI.
